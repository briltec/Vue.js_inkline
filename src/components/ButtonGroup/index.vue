<template src="./template.html" />
<script src="./script.js" />
<style src="../Button/style.styl" lang="stylus" scoped />
<style src="./style.styl" lang="stylus" scoped />
