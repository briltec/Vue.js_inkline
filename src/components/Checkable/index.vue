<script src="./script.js" />
<!--<style src="./style.styl" lang="stylus" />-->
