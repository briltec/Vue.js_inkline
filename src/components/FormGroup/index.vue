<template src="template.html" />
<script src="script.js" />
<style src="../Input/style.styl" lang="stylus" scoped />
<style src="style.styl" lang="stylus" scoped />
