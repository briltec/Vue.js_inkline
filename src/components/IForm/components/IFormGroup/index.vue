<template src="./template.html" />
<style src="./style.scss" lang="scss" />
<script src="./script.js" />
