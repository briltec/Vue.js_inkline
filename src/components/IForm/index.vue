<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="../IInput/style.scss" lang="scss" scoped />
<style src="./style.scss" lang="scss" scoped />
