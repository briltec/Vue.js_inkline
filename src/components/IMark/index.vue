<template src="./template.html" />
<script src="./script.ts" lang="ts" />
