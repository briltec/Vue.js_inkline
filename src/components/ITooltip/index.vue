<template src="./template.html" />
<script src="./script.ts" lang="ts" />
<style src="./style.scss" lang="scss" />
