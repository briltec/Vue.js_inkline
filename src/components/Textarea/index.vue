<template src="./template.html" />
<style src="../Input/style.styl" lang="stylus" />
<script src="./script.js" />
