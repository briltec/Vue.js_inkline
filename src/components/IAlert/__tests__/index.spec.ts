import { shallowMount } from '@vue/test-utils';
import { IAlert } from '@inkline/inkline/components';

describe('Components', () => {
    describe('IAlert', () => {
        const props = {
            color: 'primary',
            size: 'md'
        };

        it('should be named correctly', () => {
            expect(IAlert.name).toEqual('IAlert');
        });

        it('should render correctly', () => {
            const wrapper = shallowMount(IAlert, { props });
            expect(wrapper.html()).toMatchSnapshot();
        });

        describe('computed', () => {
            describe('classes', () => {
                it('should return classes object', () => {
                    const wrapper: any = shallowMount(IAlert, { props });

                    expect(wrapper.vm.classes).toEqual({
                        [`-${wrapper.vm.color}`]: true,
                        [`-${wrapper.vm.size}`]: true,
                        '-dismissible': false,
                        '-with-icon': false
                    });
                });
            });
        });

        describe('watch', () => {
            describe('modelValue', () => {
                it('should update dismissed', async () => {
                    const wrapper: any = shallowMount(IAlert, {
                        props: {
                            modelValue: true,
                            ...props
                        }
                    });

                    expect(wrapper.vm.dismissed).toEqual(false);
                    wrapper.setProps({ modelValue: false });
                    await wrapper.vm.$nextTick();
                    expect(wrapper.vm.dismissed).toEqual(true);
                });
            });
        });

        describe('methods', () => {
            describe('dismiss()', () => {
                it('should emit update:modelValue with false', () => {
                    const wrapper: any = shallowMount(IAlert);

                    wrapper.vm.dismiss();

                    expect(wrapper.emitted()['update:modelValue'][0]).toEqual([false]);
                });
            });
        });
    });
});
