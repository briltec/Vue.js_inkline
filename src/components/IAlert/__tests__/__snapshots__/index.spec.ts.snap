// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components IAlert should render correctly 1`] = `
<div class="alert -primary -md">
  <!--v-if-->
  <div class="content"></div>
  <!--v-if-->
</div>
`;
